### PR TITLE
A few fixes

### PR DESCRIPTION
**Pull request by not-an-aardvark**
'Eggcup Egg' is now called 'Egg Cup'. Resolves the unrelated issues #367 and #368.

### CODE REVIEW

File: assets/js/userCtrl.js [hunk 1]
```diff
@@ -106,34 +107,7 @@ module.exports = function ($scope, $filter, $location, $timeout) {
   };
 
   $scope.formattedName = function (name) {
```
Comment (YaManicKill) [inline assets/js/userCtrl.js]: Just bind this directly, no?

$scope.formattedName = flairsService.formattedName

File: assets/js/userCtrl.js [hunk 2]
```diff
@@ -3,6 +3,7 @@ var io = require("sails.io.js")(socket);
 var regex = require("regex");
 var _ = require("lodash");
 var $ = require("jquery");
+var flairsService = require("../../api/services/Flairs.js");
```
Comment (YaManicKill) [inline assets/js/userCtrl.js]: flairService, please. That plural really grinds with me.

File: assets/js/refCtrl.js [hunk 3]
```diff
@@ -162,7 +162,7 @@ module.exports = function ($scope, $filter) {
   };
 
   $scope.getFlairTextForSVEx = function () {
-    if (!$scope.refUser || !$scope.refUser.flair) {
+    if (!$scope.refUser || !$scope.refUser.flair || !$scope.refUser.flair.svex || !$scope.refUser.flair.svex.flair_css_class) {
```
Comment (not-an-aardvark) [inline assets/js/refCtrl.js]: Prevents client error if $scope.refUser.flair.svex.flair_css_class is undefined.

File: api/controllers/AuthController.js [hunk 4]
```diff
@@ -78,8 +78,11 @@ module.exports = {
                   console.log('Failed to give /u/' + user.name + ' moderator status');
                   return res.view(500, {error: "You appear to be a mod, but you weren't given moderator status for some reason.\nTry logging in again."});
                 }
-                // Redirect to the mod authentication page. If the state ends in '_modlogin', the user was just there, so don't redirect there again.
+                /* Redirect to the mod authentication page. If the state ends in '_modlogin', the user was just there, so get rid of the _modlogin flag
+                *  instead of redirecting there again. If a mod ends up on a different page while they still have the _modlogin flag, they have not
+                *  successfully authenticated, so they will get redirected to /auth/modauth. */
                 if (req.session.state.substr(-9) === '_modlogin') {
+                  req.session.state = req.session.state.slice(0,-9);
```
Comment (not-an-aardvark) [inline api/controllers/AuthController.js]: See #368 for further details.

**Pull request merged**